packages/core/src/editor/query.tsx

```diff
@@ -152,14 +140,30 @@ export function QueryMethods(Editor: EditorState) {
 
       return output;
     },
+
+    /**
+     * Get the current Editor options
+     */
+    getOptions(): Options {
+      return options;
+    },
+
+    getState(): Record<NodeId, SerializedNodeData> {
```

Comment:
Same as `deserialize` - since we're not actually get the actual `EditorState`, I feel this might cause confusion. Maybe, let it remain as `serialize`, with a parameter to specify if the return value should be in JSON? Although I feel we should omit the JSON capability in the future since it's pretty trivial, but for now it might help those who are expecting the value to be in JSON.

Reply:
yeah I also feel the `getState` name is a bit confusing.

Most of the methods in `query` though have a getter notation, so I think we should also follow a `get____` something pattern like the rest of the getters.

Maybe `getSerializedNodes`? I'm not in love with it though, it's a mouthful. Also, if we call it`serialize` we'll never be able to deprecate the actual `serialize` method

Reply:
Hm, yeah but it is the quite descriptive haha. I don't mind `getSerializedNodes` for now with a deprecation notice on `serialize`